salt/modules/heat.py

```diff
@@ -0,0 +1,870 @@
+        # Optional
+        keystone.region_name: 'RegionOne'
+
+    If configuration for multiple OpenStack accounts is required, they can be
+    set up as different configuration profiles:
+    For example::
+
+        openstack1:
+          keystone.user: admin
+          keystone.password: verybadpass
+          keystone.tenant: admin
+          keystone.auth_url: 'http://127.0.0.1:5000/v2.0/'
+
+        openstack2:
+          keystone.user: admin
+          keystone.password: verybadpass
+          keystone.tenant: admin
+          keystone.auth_url: 'http://127.0.0.2:5000/v2.0/'
+
+    With this configuration in place, any of the heat functions can make use of
+    a configuration profile by declaring it explicitly.
+    For example::
+
+        salt '*' heat.flavor_list profile=openstack1
+'''
+
+# Import Python libs
+from __future__ import absolute_import
+import time
+import json
+import logging
+# Import third party libs
+import yaml
+# Import salt libs
+import salt.ext.six as six
+import salt.utils
+from salt.exceptions import SaltInvocationError
+
+# pylint: disable=import-error
+HAS_HEAT = False
+try:
+    from heatclient import client
+    from heatclient import exc
+    HAS_HEAT = True
+except ImportError:
+    pass
+
+HAS_OSLO = False
+try:
+    from oslo_serialization import jsonutils
+    HAS_OSLO = True
+except ImportError:
+    pass
+
+SECTIONS = (
+    PARAMETER_DEFAULTS, PARAMETERS, RESOURCE_REGISTRY, EVENT_SINKS
+) = (
+    'parameter_defaults', 'parameters', 'resource_registry', 'event_sinks'
+)
+
+if hasattr(yaml, 'CSafeLoader'):
+    YamlLoader = yaml.CSafeLoader
+else:
+    YamlLoader = yaml.SafeLoader
+
+if hasattr(yaml, 'CSafeDumper'):
+    YamlDumper = yaml.CSafeDumper
+else:
+    YamlDumper = yaml.SafeDumper
+
+
+def _represent_yaml_str(self, node):
+    '''
+    Represent for yaml
+    '''
+    return self.represent_scalar(node)
+YamlDumper.add_representer(u'tag:yaml.org,2002:str',
+                           _represent_yaml_str)
+YamlDumper.add_representer(u'tag:yaml.org,2002:timestamp',
+                           _represent_yaml_str)
+
+
+def _construct_yaml_str(self, node):
+    '''
+    Construct for yaml
+    '''
+    return self.construct_scalar(node)
+YamlLoader.add_constructor(u'tag:yaml.org,2002:str',
+                           _construct_yaml_str)
+YamlLoader.add_constructor(u'tag:yaml.org,2002:timestamp',
+                           _construct_yaml_str)
+
+logging.basicConfig(level=logging.DEBUG)
+log = logging.getLogger(__name__)
+
+
+def __virtual__():
+    '''
+    Only load this module if heat
+    is installed on this minion.
+    '''
+    if HAS_HEAT and HAS_OSLO:
+        return 'heat'
+    return (False, 'The heat execution module cannot be loaded: '
+                   'the heatclient and oslo_serialization'
+                   # 'the heatclient and keystoneclient and oslo_serialization'
+                   ' python library is not available.')
+
+
+__opts__ = {}
```

Comment:
Hrm, OK. We'll get this in then and I'll make a note to go see if we can chase down the reason for that. Thanks for letting me know.